### PR TITLE
OpenAI quickstarts

### PR DESCRIPTION
**Pull request by diberry**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by diberry** [comment]
@deyaaeldeen Please review for the TS version of the OpenAI Completions quickstart.

File: quickstarts/completions/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "completions",
+  "version": "1.0.0",
+  "main": "index.js",
+  "scripts": {
+    "key": "ts-node src/index-key.ts",
+    "no-key": "ts-node src/index-no-key.ts",
+    "build": "tsc",
```
Comment (deyaaeldeen) [inline quickstarts/completions/package.json:8]: Should `typescript` be added as a dev dependency?

File: quickstarts/completions/src/index-key.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { AzureOpenAI } from "openai";
+import { type Completion } from "openai/resources/index";
+import "dotenv/config";
+
+// You will need to set these environment variables or edit the following values
+const endpoint = process.env["AZURE_OPENAI_ENDPOINT"] || "<endpoint>";
+const apiKey = process.env["AZURE_OPENAI_API_KEY"] || "<api key>";
+
+// Azure OpenAI API version and deployment
+const apiVersion = "2024-04-01-preview";
```
Comment (deyaaeldeen) [inline quickstarts/completions/src/index-key.ts:10]: nitpick: use a new API version.
```suggestion
const apiVersion = "2024-07-01-preview";

```

File: quickstarts/completions/src/index-key.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+function getClient(): AzureOpenAI {
+  return new AzureOpenAI({ endpoint, apiKey, apiVersion, deployment });
+}
+async function getCompletion(
+  client: AzureOpenAI,
+  prompt: string[],
+  model: string,
+  max_tokens: number
+): Promise<Completion> {
+  return await client.completions.create({
```
Comment (deyaaeldeen) [inline quickstarts/completions/src/index-key.ts:26]:
```suggestion
  return client.completions.create({

```

File: quickstarts/completions/src/index-key.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    prompt,
+    model,
+    max_tokens,
+  });
+}
+async function getChoices(completion: Completion): Promise<void> {
```
Comment (deyaaeldeen) [inline quickstarts/completions/src/index-key.ts:32]: Should it be printChoices?
```suggestion
async function printChoices(completion: Completion): Promise<void> {

```

File: quickstarts/completions/src/index-key.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+const deployment = "gpt-35-turbo-instruct";
+
+// Chat prompt and max tokens
+const prompt = ["When was Microsoft founded?"];
+const maxTokens = 128;
+
+function getClient(): AzureOpenAI {
+  return new AzureOpenAI({ endpoint, apiKey, apiVersion, deployment });
+}
+async function getCompletion(
```
Comment (deyaaeldeen) [inline quickstarts/completions/src/index-key.ts:20]: Is there a reason to have these simple functions over simply inlining them?

File: quickstarts/completions/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  "compilerOptions": {
+    "module": "NodeNext",
+    "target": "ESNext",
+    "moduleResolution": "NodeNext",
+
+    "outDir": "dist",
+    "rootDir": "./src",
+
+    "esModuleInterop": true,
+    "skipLibCheck": true
```
Comment (deyaaeldeen) [inline quickstarts/completions/tsconfig.json:11]: are these needed?

File: quickstarts/completions/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  },
+  "include": ["src/**/*"],
+  "exclude": ["node_modules", "dist"]
```
Comment (deyaaeldeen) [inline quickstarts/completions/tsconfig.json:14]: this is not needed I belive

File: quickstarts/completions/.eslintrc [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (deyaaeldeen) [inline quickstarts/completions/.eslintrc:1]: should we add eslint as a dev dependency?

File: quickstarts/completions/src/index-key.ts (hunk 5 above)
Comment (diberry) [reply quickstarts/completions/src/index-key.ts:20]: @deyaaeldeen The purpose is provide reusable typed code in the quickstart documentation.

File: quickstarts/completions/.eslintrc (hunk 8 above)
Comment (diberry) [reply quickstarts/completions/.eslintrc:1]: done

File: quickstarts/openai/src/text-to-speech-key.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+function getClient(): AzureOpenAI {
+  return new AzureOpenAI({ endpoint, apiKey, apiVersion, deployment });
+}
+
+export async function main() {
+  console.log("== Text to Speech Sample ==");
+
+  const client = getClient();
+  const response = await client.audio.speech.create({
+    model: deployment,
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/text-to-speech-key.ts:41]:
```suggestion
    model: "",

```

File: quickstarts/openai/src/text-to-speech-no-key.ts [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  );
+  return new AzureOpenAI({ azureADTokenProvider, deployment: deployment, apiVersion });
+}
+
+export async function main() {
+  console.log("== Text to Speech Sample ==");
+
+  const openai = getClient();
+  const response = await openai.audio.speech.create({
+    model: deployment,
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/text-to-speech-no-key.ts:42]:
```suggestion
    model: "",

```

File: quickstarts/openai/src/text-to-speech-key.ts [hunk 11]
```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to convert text into speech.
+ *
+ * @summary text to speech.
+ */
+
+import "openai/shims/node";
```
[the remark this answers is not in the document]
Comment (diberry) [reply quickstarts/openai/src/text-to-speech-key.ts:10]: Sure.

File: quickstarts/openai/src/text-to-speech-no-key.ts [hunk 12]
```diff
@@ -1,56 +1,57 @@
...
+  const scope = "https://cognitiveservices.azure.com/.default";
+  const azureADTokenProvider = getBearerTokenProvider(
+    new DefaultAzureCredential(),
+    scope
+  );
+  return new AzureOpenAI({
+    azureADTokenProvider,
+    deployment: deploymentName,
+    apiVersion,
+  });
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/text-to-speech-no-key.ts:29]: should we pass the endpoint similar to the other samples?

File: quickstarts/openai/src/text-to-speech-no-key.ts [hunk 13]
```diff
@@ -1,56 +1,57 @@
...
+  const response = await client.audio.speech.create(params);
+  if (response.ok) return response.body;
+  throw new Error(`Failed to generate audio stream: ${response.statusText}`);
+}
+export async function main() {
+  console.log("== Text to Speech Sample ==");
+
+  const client = getClient();
+  const streamToRead = await generateAudioStream(client, {
+    model: deploymentName,
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/text-to-speech-no-key.ts:45]: `model: ""`

File: quickstarts/openai/src/use-your-data-no-key.ts [hunk 14]
```diff
@@ -1,80 +1,80 @@
...
+  const scope = "https://cognitiveservices.azure.com/.default";
+  const azureADTokenProvider = getBearerTokenProvider(
+    new DefaultAzureCredential(),
+    scope
+  );
+  return new AzureOpenAI({
+    azureADTokenProvider,
+    deployment: deploymentName,
+    apiVersion,
+  });
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/use-your-data-no-key.ts:27]: same here, should we pass the endpoint similar to the other samples?

File: quickstarts/openai/src/vision-no-key.ts [hunk 15]
```diff
@@ -1,32 +1,32 @@
...
   const azureADTokenProvider = getBearerTokenProvider(
     new DefaultAzureCredential(),
     scope
   );
-  return new AzureOpenAI({ azureADTokenProvider, deployment: deploymentName, apiVersion });
+  return new AzureOpenAI({
+    azureADTokenProvider,
+    deployment: deploymentName,
+    apiVersion,
+  });
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/vision-no-key.ts:29]: same here, should we pass the endpoint similar to the other samples?

File: quickstarts/openai/src/whisper-no-key.ts [hunk 16]
```diff
@@ -1,35 +1,43 @@
...
+  const scope = "https://cognitiveservices.azure.com/.default";
+  const azureADTokenProvider = getBearerTokenProvider(
+    new DefaultAzureCredential(),
+    scope
+  );
+  return new AzureOpenAI({
+    azureADTokenProvider,
+    deployment: deploymentName,
+    apiVersion,
+  });
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/whisper-no-key.ts:26]: same here, should we pass the endpoint similar to the other samples?

File: quickstarts/openai/src/dall-e-no-key.ts [hunk 17]
```diff
@@ -1,44 +1,48 @@
...
+  const scope = "https://cognitiveservices.azure.com/.default";
+  const azureADTokenProvider = getBearerTokenProvider(
+    new DefaultAzureCredential(),
+    scope
+  );
+  return new AzureOpenAI({
+    azureADTokenProvider,
+    deployment: deploymentName,
+    apiVersion,
+  });
```
Comment (deyaaeldeen) [inline quickstarts/openai/src/dall-e-no-key.ts:26]: same here, should we pass the endpoint similar to the other samples?

**Pull request merged**